src/OrchardCore.Modules/OrchardCore.Media.Azure/MediaBlobStorageOptionsCheckFilter.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Admin;
+using OrchardCore.DisplayManagement.Notify;
+
+namespace OrchardCore.Media.Azure
+{
+    public class MediaBlobStorageOptionsCheckFilter : ActionFilterAttribute
+    {
+        private readonly MediaBlobStorageOptions _options;
+        private readonly ILogger<MediaBlobStorageOptionsCheckFilter> _logger;
+        private readonly INotifier _notifier;
+        private readonly IHtmlLocalizer<MediaBlobStorageOptionsCheckFilter> T;
+
+        public MediaBlobStorageOptionsCheckFilter(
+            IOptions<MediaBlobStorageOptions> options, 
+            ILogger<MediaBlobStorageOptionsCheckFilter> logger, 
+            INotifier notifier, 
+            IHtmlLocalizer<MediaBlobStorageOptionsCheckFilter> localizer)
+        {
+            _options = options.Value;
+            _logger = logger;
+            _notifier = notifier;
+            T = localizer;
+        }
+
+        public override void OnActionExecuted(ActionExecutedContext context)
+        {
+            if (!CheckOptions(_options.ConnectionString, _options.ContainerName, _logger) && AdminAttribute.IsApplied(context.HttpContext))
```

Comment:
If the service is a singleton (which I assume it can be), we can memoize the the result of this method.

Reply:
Yes, we could but I don't think it's worth it. 3 string comparisons, I doubt you can ever measure the impact - overhead of memoization is most probably higher.